### PR TITLE
Feature/reaction voting

### LINKED ISSUE #251: Reaction Based Voting
Provide a mechanism for Mods/Admins/Users (potentially separating privileges by commands) to vote via reactions.

Could work in one of two ways:

1. In a public channel (bot will need to remove any invalid reaction votes)
2. In a channel where Users cannot send messages

Command syntax could be something like the following:
!startvote Message | option 1, :one: | option 2, :two: | option 3, :three:

### PR DESCRIPTION
**Pull request by chrislennon**
closes #251 

Please give this a test - I will also probably have to tidy up some sloppy work too.

I don't believe this will cover server [specific emoji](https://discord.js.org/#/docs/main/stable/class/Guild?scrollTo=emojis) yet.

### CODE REVIEW

File: commands/startvote.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  allowDM: false,
+  process: (bot, message) => {
+
+    function generateVote(message) {
+
+      let guild = bot.guilds.first();
+      let msg = message.cleanContent;
+
+      // TODO handle removal of command prefix bettwe
+      msg = msg.slice(11);
```
Comment (jmariondev) [inline commands/startvote.js]: Where is this magic value coming from?

File: commands/startvote.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+module.exports = {
+  usage: 'Message | option 1, :one: | option 2, :two: | option n, :shrug:',
+  description: 'Start a reaction based vote!',
+  allowDM: false,
+  process: (bot, message) => {
+
+    function generateVote(message) {
+
+      let guild = bot.guilds.first();
```
Comment (jmariondev) [inline commands/startvote.js]: Unused

File: commands/startvote.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        let emoji = optParts[1].trim();
+        outputString += item + ': ' + emoji + '\n';
+        outputReactions.push(emoji);
+      }
+      outputString += '```';
+
+      return outputString;
+    }
+
+    let outputReactions = [];
```
Comment (jmariondev) [inline commands/startvote.js]: `const` :wank:

(various other `lets` as well)

**Review by jmariondev — DISMISSED** [review]
Here's a couple things just from reviewing the diff. I can test this more properly at a later time.

I like this idea!

File: commands/startvote.js (hunk 1 above)
Comment (chrislennon) [reply commands/startvote.js]: Yeah... 11 is the length of the `!startvote ` prefix. Should probably be a derived value.

**Comment by otter-computer** [comment]
I've given this PR a good ol' refactor. Stripped out a lot of the logic for string operations and given a standard set of emoji to use for voting. Currently multi-option votes are disabled, only allowing for a Boolean 👍 👎 vote (mainly because Windows 10 doesn't have emoji chars for the numbers I want to use and it was an absolute pain...).

I will close current issue and create a new one for pending multi-vote feature.

**Review by otter-computer — APPROVED** [review]

**Comment by chrislennon** [comment]
👍 changes look good

**Pull request merged**